x-pack/plugins/security_solution/server/endpoint/routes/actions/list_handler.ts

```diff
@@ -31,6 +33,10 @@ const formatCommandValues = (
   value: ResponseActionsApiCommandNames | ResponseActionsApiCommandNames[] | undefined
 ): undefined | ResponseActionsApiCommandNames[] => (typeof value === 'string' ? [value] : value);
 
+const formatAgentTypeValues = (
+  value: ResponseActionAgentType | ResponseActionAgentType[] | undefined
+): undefined | ResponseActionAgentType[] => (typeof value === 'string' ? [value] : value);
```

Comment:
instead of adding a 3rd function for the same functionality, I'd suggest to use a generic function:
```ts
const formatRequestParams = <T extends string>(value: T | T[] | undefined): T[] | undefined =>
  typeof value === 'string' ? [value] : value;
```

(actually the function `formatStatusValues` is also the same, only it does not accept `undefined`, but I couldn't come up a generic covering that, too)

Reply:
Yeah I was in the process of doing this actually. Thanks for suggesting the types here.

Reply:
Updated